### PR TITLE
Populate cursorAfter in datastore v1beta2

### LINKED ISSUE #548: Lazy loading of entities when running a query
I just want to scroll all entities of a specific kind in a lazy way. I have tried to add a limit but as intended I have only the number of specified results and not the entire dataset.

```java
StructuredQuery<Entity> query = Query.entityQueryBuilder()
    .kind("user")
    .limit(100)
    .build();

QueryResults<Entity> results = datastore.run(query);
```

How to do it ? Did you plan to add a `fetchSize()` parameter ?

Without `limit()` I have a DatastoreException: Server returned an error INTERNAL 500

### PR DESCRIPTION
**Pull request by ajkannan**
Fixes #548

### CODE REVIEW

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResults.java [hunk 1]
```diff
@@ -35,8 +35,9 @@
   Class<?> resultClass();
 
   /**
-   * Returns the Cursor for point after the value returned in the last {@link #next} call.
-   * Not currently implemented (depends on v1beta3).
+   * Returns the Cursor for the point after the value returned in the last {@link #next} call. Until
+   * v1beta3 is supported, this field should only be used if you have set a limit and the number of
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResults.java]: I don't think `v1beta3` says much to people. Instead I think we should just say "currently cursorAfter returns null in all cases but the last result" or something like that.

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResultsImpl.java [hunk 2]
```diff
@@ -100,6 +100,6 @@ protected T computeNext() {
   @Override
   public Cursor cursorAfter() {
     //return new Cursor(cursor); // only available in v1beta3
-    return null;
+    return new Cursor(queryResultBatchPb.getEndCursor());
```
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResultsImpl.java]: I think we should set it to non-null only after last result is consumed.

File: gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/DatastoreTest.java [hunk 3]
```diff
@@ -463,6 +466,95 @@ public void testRunStructuredQuery() {
...
+    EasyMock.expect(rpcFactoryMock.create(EasyMock.anyObject(DatastoreOptions.class)))
+        .andReturn(rpcMock);
+    List<RunQueryResponse> responses = buildResponsesForQueryPaginationWithLimit();
+    for (int i = 0; i < responses.size(); i++) {
+      EasyMock.expect(rpcMock.runQuery(EasyMock.anyObject(RunQueryRequest.class)))
+          .andReturn(responses.get(i));
+    }
+    EasyMock.replay(rpcFactoryMock, rpcMock);
+    Datastore mockDatastore =
+        this.options.toBuilder()
```
Comment (aozarov) [inline gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/DatastoreTest.java]: Nit "is "this." necessary?

**Comment by ajkannan** [comment]
Added commit to address comments

**Comment by ajkannan** [comment]
@mziccard In [663e54b](https://github.com/ajkannan/gcloud-java/commit/663e54b356162b6890fd8175c9cc17a4a70dd097) I added a little bit more to the documentation for `QueryResults`.  Do you think there's anything else that should be mentioned to make it clearer?

**Comment by aozarov** [comment]
Looks good. @mziccard I am merging it but please comment and we should open a separate PR for any further doc changes.

**Pull request merged**